simulation/simulation.py

```diff
@@ -130,95 +152,261 @@ def distance(ax, ay, bx, by):
     # GREAT CIRCLE - use for lat/long, fast and inaccurate
 
 
+class RogueCar:
+    @classmethod
+    async def create_rogue(cls, starttime, loc, dest, manager):
+        rogue = cls(starttime, loc, dest, manager)
+        await rogue.tried[0].register(rogue.first_attempt)
+        return rogue
+
+    def __init__(self, starttime, loc, dest, manager):
+        self.startX = loc.latitude
+        self.startY = loc.longitude
+        self.destX = dest.latitude
+        self.destY = dest.longitude
+        self.waypoints = []
+        self.drawing = True
+        self.starttime = starttime
+        self.bestLot = None
+        self.tried = []
+        self.manager = manager
+        self.first_attempt = None
+
+        closeness = 5
+        self.speed = 6
+        stupidity = 0.5
+
+        self.waypoints.append(Waypoint(starttime, self.startX, self.startY))
+
+        currentX = self.startX
+        currentY = self.startY
+
+        lasttime = time.time()
+
+        while geodistance(currentX, currentY, self.destX, self.destY) > closeness:
+            # pick a random place to go next, that is probably but not definitely closer to the destination
+            mu = (currentX + self.destX) * 0.5
+            sigma = abs(currentX - self.destX) * stupidity
+            newX = random.normalvariate(mu, sigma)
+
+            mu = (currentY + self.destY) * 0.5
+            sigma = abs(currentY - self.destY) * stupidity
+            newY = random.normalvariate(mu, sigma)
+
+            # distance = math.sqrt((currentX - newX)**2 + (currentY - newY)**2)
+            distance = geodistance(currentX, currentY, newX, newY)
+
+            duration = distance / self.speed
+
+            currentX = newX
+            currentY = newY
+
+            self.waypoints.append(Waypoint(lasttime + duration, newX, newY))
+
+            lasttime += duration
+
+        bestLot = None
+        bestDistance = 10000000000000
+        # once close enough to the destination, find the nearest parking lot
+        for ilot in manager.lots:
+            currentDistance = geodistance(currentX, currentY, ilot.lot.location.latitude, ilot.lot.location.longitude)
+            if currentDistance < bestDistance:
+                bestDistance = currentDistance
+                bestLot = ilot
```

Comment:
Maybe this is too optimistic for rogue cars? They might not know the location of nearby lots this accurately